code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -37,6 +37,55 @@
 	icon_resting = "cat_rest"
 	gender = FEMALE
 	gold_core_spawnable = CHEM_MOB_SPAWN_INVALID
+	var/list/family = list()
+	var/lives = 9
+	var/memory_saved = 0
+	var/list/children = list() //Actual mob instances of children
+	var/cats_deployed = 0
+
+/mob/living/simple_animal/pet/cat/Runtime/New()
+	Read_Memory()
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/Life()
+	if(!cats_deployed && ticker.current_state >= GAME_STATE_SETTING_UP)
+		Deploy_The_Cats()
+	if(!stat && ticker.current_state == GAME_STATE_FINISHED && !memory_saved)
+		Write_Memory()
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/death()
+	if(!memory_saved)
+		Write_Memory(1)
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Read_Memory()
+	var/savefile/S = new /savefile("data/npc_saves/Runtime.sav")
+	S["family"] 			>> family
+	if(isnull(family))
+		family = list()
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Write_Memory(dead)
+	var/savefile/S = new /savefile("data/npc_saves/Runtime.sav")
+	if(dead)
+		S["lives"] 				<< lives - 1
+	family = list()
+	for(var/mob/living/simple_animal/pet/cat/C in mob_list)
+		if(istype(C,type) || C.stat || !C.butcher_results || C.name == "SyndiCat")

```

Comment:
This won't need the SyndiCat check, since it will only choose from Runtime's children anyway. TG also checks for a non-zero Z-level here, perhaps to skip kittens hidden in lockers and such.